metadata-ingestion/tests/unit/test_transform_dataset.py

```diff
@@ -775,14 +775,19 @@ def test_simple_dataset_tags_transformation(mock_time):
             ]
         )
     )
-    assert len(outputs) == 3
+
+    assert len(outputs) == 5
 
     # Check that tags were added.
     tags_aspect = outputs[1].record.aspect
     assert tags_aspect
     assert len(tags_aspect.tags) == 2
     assert tags_aspect.tags[0].tag == builder.make_tag_urn("NeedsDocumentation")
 
+    # Check new tag entity should be there
+    assert outputs[2].record.entityUrn == builder.make_tag_urn("NeedsDocumentation")
+    assert outputs[3].record.entityUrn == builder.make_tag_urn("Legacy")
```

Comment:
can you add a test that shows the handle_end_of_stream method getting called on a LegacyMCETransformer?

Reply:
We don't have LegacyMCETransformer implementation. All transformers are of type SingleAspectTransformer.